nservicebus/azure-service-bus/performance-tuning_body_asb_[7,).partial.md

```diff
@@ -0,0 +1,61 @@
+  * `NumberOfClientsPerEntity()`: Keep this value equal to the number of factories, this ensures there is 1 receive client per factory.
+
+Other settings that impact performance a lot, but their values differ quite a bit per scenario, are:
+
+  * `LimitMessageProcessingConcurrencyTo()`: This is the global concurrency limit across all receive entities. Ensure this value matches the per receiver desired concurrency for the scenario, multiplied by the number of receivers.
+  * `MessageReceivers().PrefetchCount()`: Value is per receiver, in general the following rule applies: The lower the throughput rate of an individual receiver, the lower this value should be.
+  * `Transactions()`: The most significant performance impact comes from the transactional guarantees needed, `SendsAtomicWithReceive` will offer less performance then `ReceiveOnly`.
+
+### Sends Atomic With Receive
+
+In order to guarantee that sends occur as an atomic operation with the receive confirmation, the ASB SDK requires that both operations are enclosed in a `Serializable` transaction. This implies that:
+
+  * All atomic operations are executed in a 'Serial' fashion, aka 1 by 1, therefore it does not make sense to allow many concurrent operations per client, nor does it make much sense to prefetch a lot of messages.
+	  - Concurrency: 2 or 4 per client seems to work best
+	  - PrefetchCount: 20 or less seem to be good values
+  * Setting up the transactional connection takes up quite some time, this causes the first operational minutes to perform very 'bumpy', but after a while the throughput becomes consistent.
+  * All send operations that occur in the scope of a single receive (inside a handler), must fit into the transaction. Therefore a maximum of 100 messages can be sent this way. The good news is that doing so has little additional implications on performance tuning as 1 send operation or 100 is actually the same thing except for the size of the operation.
+
+### Receive only
+
+This mode has a lot less constraints and will work better with high numbers for concurrency and prefetch count, which should result in a pretty high throughput (likely over 1000/s even for remote connections)
+
+  * Concurrency: Values in the range of 128 per receive client seems to work well.
+  * PrefetchCount: Set this to 1x or 2x the per receiver concurrency, so 128 or 256.
+
+But, sending messages out at the same time does influence this mode more. If the average handler sends out only a single message then the above values still hold true. However if the handler sends out more messages, let's say 10 or 100 messages per receive, the concurrency and prefetchcount should be dialed down to allow more bandwidth to the senders.
```

Comment:
> If the average handler sends out only a single message then the above values still hold true. However if the handler sends out more messages, let's say 10 or 100 messages per receive, the concurrency and prefetchcount should be dialed down to allow more bandwidth to the senders.

What does this mean? Why reduce the prefetching/concurrency because handlers are chatty? its data going out instead of in.

I can imagine that if these are very large messages that in the beginning ingress could be hindering exgress but once the prefect buffer is warmed up it should not really matter.